Automattic/wp-calypso
# Themes Discovery: Add filter based carousels to the Discovery view.

In this PR we implement the theme carousel collections that rely on existing Categories.

Related to 
- https://github.com/Automattic/dotcom-forge/issues/4237

Builds on (the following need to be merged before this PR)
- https://github.com/Automattic/wp-calypso/pull/83044
## Proposed Changes

* Implemented carousels for the categories detailed in the main issue.
* Refactored how theme collection components are created, now they're dynamically generated so that adding a new collection OF AN EXISTING TYPE does not require changing where the collections get instantiated.
* Refactored how the Theme Showcase calculates the title and description to show. 
* We now support any type of combination of existing filters when building a carousel since I've implemented a key that we'll use to search in the CollectionTypes map.

## Testing Instructions



* Open the Calyps live link
* go to `/themes?flags=themes/discovery-lots` 
* You should see 4 new fully functional categories.
* Make sure there are no regressions in the navigation.

Before:
![Captura de pantalla 2023-10-18 a las 13 58 46](https://github.com/Automattic/wp-calypso/assets/1989914/661d235c-e726-4811-a05c-2245cd040a18)

After:
![Captura de pantalla 2023-10-18 a las 13 59 21](https://github.com/Automattic/wp-calypso/assets/1989914/0bf68b99-2672-4fcb-801d-037fa6d233fb)


## Pre-merge Checklist
